Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -23,6 +23,10 @@ public class PressableButton : MonoBehaviour, IMixedRealityTouchHandler
         [Tooltip("The object that is being pushed.")]
         private GameObject movingButtonVisuals = null;
 
+        [SerializeField]
```

Comment:
I can see how you needed this to implement the specific behavior you needed for the shell buttons, but is it something you think should be in the generic pressable button, the thing that most developers use?

It seems like the specific behavior you want is to position the object exactly half way along the press amount. For this very specific behavior it may make more sense to have a component on the moving icon that positions itself based on the press direction of the button and the press amount.

Reply:
Since the intention of this control is to provide the shell-parity, great default experience, we would like to include it here so that developers can get it free when it is used without modification. From the customization standpoint, I agree that we can expose this as a configurable element.

Reply:
Forgot to mention that this fixes the bug of icon & text being occluded by the back plate when pressed fully. Current button has a wrong thickness dimension which made the icon & text still visible on a full press.

Reply:
Hm, I actually didn't realize that the intention of PressableButton.cs was to provide shell parity button. I thought PressableButton.cs intention was to provide a pressable button, meaning a button that has some travel amount when touched, and then fires press and release events based on thresholds. I think that might be where my  comments are coming from. I wonder if the only PressableButton behavior we provide should require to match the shell so strictly in behavior. Instead it could be better to make a  ShellParityButton.cs that extends PressableButton which would have this behavior.


For example, you could do something like the following:


```
public class HoloLens2ShellParityButton : PressableButton
{
    [SerializeField] MovingButtonIconText = null;

    protected void UpdateMovingVisualsPosition()
    {
        base.UpdateMovingVisualsPosition();
        if (movingButtonIconText != null)
        {
            Debug.Assert(initialTransform != null);
            movingButtonIconText.transform.position = initialTransform.position + WorldSpacePressDirection * currentPushDistance / 2;
        }
    }
}
```

Reply:
Now I understand. Essentially, we would like to provide a baseline PressableButton which developers can leverage it for implementing any custom buttons. PressableButtonHoloLens2 and PressableButtonRound are great examples. Might be better to rename prefabs to clarify this.

![2019-05-23 14_54_06-Unity 2018 3 8f1 Personal - HandInteractionExamples unity - MRTK-Public-Microsof](https://user-images.githubusercontent.com/13754172/58289335-ec8a4200-7d6a-11e9-8b6c-f799b0e74add.png)

Reply:
Applied the updates! (description on the main comment thread)